cmd/containerd-shim-runhcs-v1/delete.go

```diff
@@ -99,6 +101,26 @@ The delete command will be executed in the container's bundle as its cwd.
 			}
 		}
 
+		// For Host Process containers if a group name is passed as the user for the container the shim will create a
+		// temporary user for the container to run as and add it to the specified group. On container exit the account will
+		// be deleted, but if the shim crashed unexpectedly (panic, terminated etc.) then the account may still be around.
+		// The username will be the container ID so try and delete it here. The username character limit is 20, so we need to
+		// slice down the container ID a bit.
+		username := idFlag[:winapi.UserNameCharLimit]
+		usrNameUTF16, err := windows.UTF16PtrFromString(username)
+		if err == nil {
+			// Always try and delete the user, if it doesn't exist we'll get a specific error code that we can use to
+			// not log any warnings.
+			if err := winapi.NetUserDel(
```

Comment:
Can you make the autogenerated syscall functions you added to `winapi` unexported, and the add wrappers that perform the `windows.UTF16PtrFromString` and other processing from here and jobcontainers there?

Reply:
I'd thought of this also. Generally we try and export exact representations of the syscalls, but in this string heavy case it's a PITA. I'd be fine with making wrappers that take in go `string`'s and do the conversion internally. Not sure what's a great way to handle the structs though, but making the functions take in `string`'s erases 80% of the annoyance.

Reply:
From what I can see in `/internal/vmcompute/`, we wrap the strings but the structs themselves are all up to the called elsewhere, so that sounds fine to me

Reply:
I made light wrappers for these, let me know if they're alright